lib/modularity/extend.ex

```diff
@@ -334,26 +334,43 @@ defmodule Bonfire.Common.Extend do
     end
   end
 
+  @loaded_extensions_key {__MODULE__, :extensions_loaded}
   @decorate time()
   defp application_loaded?(extension) do
-    loaded_apps = loaded_applications()
-    app_names = Enum.map(loaded_apps, &elem(&1, 0))
+    extensions = :persistent_term.get(@loaded_extensions_key, %{})
 
-    Enum.member?(
-      app_names,
-      extension
-    )
+    case Map.get(extensions, extension) do
+      nil ->
+        result =
+          loaded_applications()
+          |> Enum.reduce_while(false, fn {app_name, _, _}, _acc ->
+            if app_name == extension do
+              {:halt, true}
+            else
+              {:cont, false}
+            end
+          end)
+
+        :persistent_term.put(@loaded_extensions_key, Map.put(extensions, extension, result))
```

Comment:
a problem IIRC with persistent_term is that its garbage collected so it's better to set it's value only once

Reply:
Yes, but the key will only be set the amount of times of equal to the number of unique extensions names that are being checked. I profiled the `Bonfire.UI.Social.Benchmark.render_feed/` function with this code in Bonfire.Common.Extend and this was not an issue in that analysis.